### PR TITLE
decomp: finish `mistycannon`

### PR DESCRIPTION
**Pull request by xTVaser**

### CODE REVIEW

File: goal_src/levels/common/mistycannon.gc [hunk 1]
```diff
@@ -5,47 +5,2025 @@
...
+    (spawn (-> self part2) (-> self root-override trans))
+    (ja-channel-set! 0)
+    (let ((v1-11 (-> self root-override root-prim)))
+      (set! (-> v1-11 local-sphere w) (-> self blast-radius))
+      (set! (-> v1-11 prim-core world-sphere w) (-> self blast-radius))
+      (set! (-> v1-11 collide-with) (the-as uint 924))
+      (set! (-> v1-11 prim-core collide-as) (the-as uint 256))
+      )
+    (dummy-47 (-> self root-override))
+    (let ((a1-3 (new 'stack-no-clear 'vector2w)))
```
Comment (water111) [inline goal_src/levels/common/mistycannon.gc:1082]: I think this is `overlaps-others-params`, like you found on your earlier PR.

File: goal_src/levels/common/mistycannon.gc [hunk 2]
```diff
@@ -5,47 +5,2025 @@
...
+    (when (< 0.0 f0-4)
+      (set! (-> arg0 time) (sqrtf f0-4))
+      (set! (-> arg0 speed) (/ (-> arg0 x) (* (-> arg0 time) (cos arg1))))
+      )
+    )
+  0
+  (none)
+  )
+
+(defbehavior mistycannon-find-best-solution mistycannon ((arg0 quadratic-solution))
```
Comment (water111) [inline goal_src/levels/common/mistycannon.gc:1514]: I bet this function is supposed to return an int

File: goal_src/levels/common/mistycannon.gc [hunk 3]
```diff
@@ -5,47 +5,2025 @@
...
+        (suspend)
+        )
+      )
+    (none)
+    )
+  :post
+  (the-as (function none :behavior mistycannon) rider-post)
+  )
+
+(defstate mistycannon-waiting-for-player-to-fuck-off (mistycannon)
```
Comment (water111) [inline goal_src/levels/common/mistycannon.gc:1870]: haha, very professional

**Review by water111 — COMMENTED** [review]
just a few tiny type things

File: goal_src/levels/common/mistycannon.gc (hunk 3 above)
Comment (xTVaser) [reply goal_src/levels/common/mistycannon.gc:1870]: it finally gets committed :tada:

**Review by water111 — APPROVED** [review]
changes look good, just some merge conflicts

**Pull request merged**